typescript/32-tournament/tournament.test.ts

```diff
@@ -0,0 +1,216 @@
+          'Devastating Donkeys;Allegoric Alaskans;loss',
+          'Courageous Californians;Blithering Badgers;draw',
+          'Allegoric Alaskans;Courageous Californians;win',
+        ].join('\n')
+      )
+    ).toBe(
+      [
+        'Team                           | MP |  W |  D |  L |  P',
+        'Allegoric Alaskans             |  3 |  2 |  0 |  1 |  6',
+        'Blithering Badgers             |  2 |  1 |  1 |  0 |  4',
+        'Courageous Californians        |  2 |  0 |  1 |  1 |  1',
+        'Devastating Donkeys            |  1 |  0 |  0 |  1 |  0',
+      ].join('\n')
+    )
+  })
+
+  it('ties broken alphabetically', () => {
+    expect(
+      new Tournament().tally(
+        [
+          'Courageous Californians;Devastating Donkeys;win',
+          'Allegoric Alaskans;Blithering Badgers;win',
+          'Devastating Donkeys;Allegoric Alaskans;loss',
+          'Courageous Californians;Blithering Badgers;win',
+          'Blithering Badgers;Devastating Donkeys;draw',
+          'Allegoric Alaskans;Courageous Californians;draw',
+        ].join('\n')
+      )
+    ).toBe(
+      [
+        'Team                           | MP |  W |  D |  L |  P',
+        'Allegoric Alaskans             |  3 |  2 |  1 |  0 |  7',
+        'Courageous Californians        |  3 |  2 |  1 |  0 |  7',
+        'Blithering Badgers             |  3 |  0 |  1 |  2 |  1',
+        'Devastating Donkeys            |  3 |  0 |  1 |  2 |  1',
+      ].join('\n')
+    )
+  })
+
+  it('ensure points sorted numerically', () => {
+    expect(
+      new Tournament().tally(
+        [
+          'Devastating Donkeys;Blithering Badgers;win',
+          'Devastating Donkeys;Blithering Badgers;win',
+          'Devastating Donkeys;Blithering Badgers;win',
+          'Devastating Donkeys;Blithering Badgers;win',
+          'Blithering Badgers;Devastating Donkeys;win',
+        ].join('\n')
+      )
+    ).toBe(
+      [
+        'Team                           | MP |  W |  D |  L |  P',
+        'Devastating Donkeys            |  5 |  4 |  0 |  1 | 12',
+        'Blithering Badgers             |  5 |  1 |  0 |  4 |  3',
+      ].join('\n')
+    )
+  })
+
+  it('large number of matches between two teams with long team name', () => {
+    const matches = []
+    const longTeamName = 'My very long team name geez I wonder what would happen with my table layout'
+    const shortTeamName = 'Short Team'
+
+    for (let i = 0; i < 105; i++) {
+      matches.push(`${longTeamName};${shortTeamName};win`)
+    }
+
+    expect(new Tournament().tally(matches.join('\n'))).toBe(
+      [
+        'Team                                                                        | MP |  W |  D |  L |  P',
+        `My very long team name geez I wonder what would happen with my table layout | 105 | 105 |  0 |  0 | 315`,
+        `Short Team                                                                  | 105 |  0 |  0 | 105 |  0`,
```

Comment:
Can't you see visually that the layout is still broken? Vertical lines aren't vertical at all